chat.js

```diff
@@ -203,7 +203,7 @@ Chat.loginfilter = function (user, oldUser, usertype) {
 
 class CommandContext {
 	/**
-	 * @param {{message: string, room: Room, user: User, connection: Connection, pmTarget?: User, cmd?: string, cmdToken?: string, target?: string, fullCmd?: string}} options
+	 * @param {{message: string, room: GlobalRoom | GameRoom | ChatRoom, user: User, connection: Connection, pmTarget?: User, cmd?: string, cmdToken?: string, target?: string, fullCmd?: string}} options
```

Comment:
You can define `type Room = GlobalRoom | GameRoom | ChatRoom`

Reply:
I feel like that could get really confusing? Having a `Room` thats different from the global type `Room` seems confusing (and it's only used twice in this file).

Reply:
You're supposed to replace the global type `Room`... I commented in the main pullreq.

Reply:
honestly we might want to whitelist functions that can accept a `GlobalRoom`. This one should be `Room | GlobalRoom`.